### PR TITLE
[ioctl] Refactor nodereward command in new ioctl

### LINKED ISSUE #3280: [ioctl] refactor nodereward command in new ioctl
### What would you like to be added
 - Step 1: refactor  command  [newcmd/node/nodereward.go](https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/node/nodereward.go) 
 - Step 2: append necessary unit test case in [newcmd/node/nodereward_test.go](https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/node/nodereward_test.go) 

#### Notes:
1. Use [function of client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) . 
2. [Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. `e.g.` `output.PrintError` could be replaced with `errors.Wrap`.
3. Global variables (config.ReadConfig  config.DefaultConfigFile etc.) have been wrapped into client interface. Please use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to access them if needed
4. The logic of new code should be similar to [the old one](https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/node/nodereward.go ).
5. In each unit test, the results should be checked whether they are as the expected. 

#### Example:
1. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountinfo.go
2. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountdelete.go
3. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountcreateadd.go

### Why is this needed
Unit tests couldn't be added for the old ioctl command (./ioctl/cmd)

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description

Build `nodereward` command in new ioctl, with the following note.

* Output package is deprecated, replace it with errors package.
* Replace `fmt.Println` with `cmd.Println`
* Refactor unit test to cover the modification.
* Check results in each unit test.

Fixes #3280 

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [x] TestNewNodeRewardCmd

**Test Configuration**:
- Firmware version: Ubuntu 21.04
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ioctl/newcmd/node/nodereward.go [hunk 1]
```diff
@@ -11,14 +11,14 @@ import (
 	"fmt"
 	"math/big"
 
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 	"google.golang.org/grpc/status"
 
 	"github.com/iotexproject/iotex-proto/golang/iotexapi"
```
Comment (huof6829) [inline ioctl/newcmd/node/nodereward.go:18]: in one group

File: ioctl/newcmd/node/nodereward.go [hunk 2]
```diff
@@ -33,29 +33,28 @@ var (
...
-		config.English: "Available Reward: %s IOTX   Total Reward: %s IOTX",
-		config.Chinese: "可用奖金: %s IOTX   总奖金: %s IOTX",
+		config.English: "Available Reward: %s IOTX, Total Reward: %s IOTX",
+		config.Chinese: "可用奖金: %s IOTX，总奖金: %s IOTX",
 	}
 )
 
 // NewNodeRewardCmd represents the node reward command
-func NewNodeRewardCmd(c ioctl.Client) *cobra.Command {
+func NewNodeRewardCmd(client ioctl.Client) *cobra.Command {
```
Comment (huof6829) [inline ioctl/newcmd/node/nodereward.go:42]: Please refactor as https://github.com/iotexproject/iotex-core/blob/773fb82140af8c7d4bf81e38b323e6eec3e75da7/ioctl/cmd/node/nodereward.go#L42-L66

File: ioctl/newcmd/node/nodereward.go [hunk 3]
```diff
@@ -91,28 +90,27 @@ func NewNodeRewardCmd(c ioctl.Client) *cobra.Command {
...
+					return errors.New("failed to convert string into big int")
 				}
 
 				message := rewardPoolMessage{
 					AvailableReward: util.RauToString(availableRewardRau, util.IotxDecimalNum),
 					TotalReward:     util.RauToString(totalRewardRau, util.IotxDecimalNum),
 				}
-				fmt.Println(message.String(rewardPoolMessageTranslation))
-
+				cmd.Println(message.String(rewardPoolMessageTranslation))
```
Comment (huof6829) [inline ioctl/newcmd/node/nodereward.go:106]: move line 155 to here directly like cmd.Printf("...",xx)

File: ioctl/newcmd/node/nodereward.go [hunk 4]
```diff
@@ -8,171 +8,182 @@ package node
 
 import (
 	"context"
-	"fmt"
 	"math/big"
 
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 	"google.golang.org/grpc/status"
```
Comment (huof6829) [inline ioctl/newcmd/node/nodereward.go:16]: delete space line16

File: ioctl/newcmd/node/nodereward.go [hunk 5]
```diff
@@ -8,171 +8,182 @@ package node
...
+				return errors.New("unknown command. \nRun 'ioctl node reward --help' for usage")
 			}
-			return output.PrintError(err)
+			return nil
 		},
 	}
-	return nc
 }
 
 type rewardPoolMessage struct {
```
Comment (huof6829) [inline ioctl/newcmd/node/nodereward.go:81]: delete

File: ioctl/newcmd/node/nodereward.go [hunk 6]
```diff
@@ -8,171 +8,182 @@ package node
...
-			m.AvailableReward, m.TotalReward)
-		return message
-	}
-	return output.FormatStringWithTrans(output.Result, m)
+	TotalBalance   string `json:"TotalBalance"`
+	TotalUnclaimed string `json:"TotalUnclaimed"`
+	TotalAvailable string `json:"TotalAvailable"`
 }
 
 type rewardMessage struct {
```
Comment (huof6829) [inline ioctl/newcmd/node/nodereward.go:87]: delete

**Review by huof6829 — APPROVED** [review]

**Comment by LuckyPigeon** [comment]
@huof6829 Thanks for the fix!

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**